crates/ide/src/ty/mod.rs

```diff
@@ -122,11 +141,17 @@ impl fmt::Debug for Ty {
 
 // Invariant: sorted by names.
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct Attrset(Arc<[(SmolStr, Ty, AttrSource)]>);
+pub struct Attrset {
+    named: Arc<[(SmolStr, Ty, AttrSource)]>,
```

Comment:
`named`'s dual is `unnamed` which usually refers to tuples' fields. But here they mean explicit labeled fields.

I think simply `fields` fits here. `rest` behaves more like constraints instead of regular fields, especially if we implement row-polymorphism in the future.